Mathlib/FieldTheory/RatFunc.lean

```diff
@@ -1761,6 +1771,30 @@ theorem coe_X : ((X : RatFunc F) : LaurentSeries F) = single 1 1 := by
 set_option linter.uppercaseLean3 false in
 #align ratfunc.coe_X RatFunc.coe_X
 
+theorem single_pow {R : Type _} [Ring R] (n : ℕ) :
+    single (n : ℤ) (1 : R) = single (1 : ℤ) 1 ^ n := by
+  induction' n with n h_ind
+  · simp only [Nat.zero_eq, Int.ofNat_eq_coe, zpow_zero]
+    rfl
+  · rw [← Int.ofNat_add_one_out, pow_succ', ← h_ind, HahnSeries.single_mul_single, one_mul,
+      add_comm]
+
+theorem single_inv (d : ℤ) {α : F} (hα : α ≠ 0) :
+    single (-d) (α⁻¹ : F) = (single (d : ℤ) (α : F))⁻¹ := by
+  apply eq_inv_of_mul_eq_one_right
+  rw [HahnSeries.single_mul_single, add_right_neg, mul_comm,
+    inv_mul_cancel hα]
+  rfl
+
+
+theorem single_zpow (n : ℤ) :
+    single (n : ℤ) (1 : F) = single (1 : ℤ) 1 ^ n := by
+  induction' n with n_pos n_neg
+  · apply single_pow
+  · rw [Int.negSucc_coe, Int.ofNat_add, Nat.cast_one, ← inv_one,
+      single_inv (n_neg + 1 : ℤ) one_ne_zero, zpow_neg, ← Nat.cast_one, ← Int.ofNat_add,
+      Nat.cast_one, inv_inj, zpow_natCast, single_pow, inv_one]
+
```

Comment:
All of these are only about `HahnSeries`, right? Could you move them to an appropriate file in `RingTheory/HahnSeries`?

Reply:
I would be happy to do so, but I think this is somewhat connected to your comment about splitting the file (which I agree with). My main question comes from the fact that Laurent series are defined as an abbreviation of Hahn Series for the monoid Z and so I tend to consider that everything pertaining to Laurent series actually has to do to Hahn ones. Would it be possible to 
1. Merge this PR as it is (modulo other remarks about proofs/naming/etc)
2. We discuss the rationale beyond splitting this and possibly other files touching Laurent series
3. I open a new PR where I split the file?

Reply:
Okay, sounds good to me!